lib/gen_stage.ex

```diff
@@ -167,13 +167,28 @@ defmodule GenStage do
   you subscribe all of them, demand will start flowing upstream and
   events downstream.
 
+  ## Demand
+
   When implementing consumers, we often set the `:max_demand` and
   `:min_demand` on subscription. The `:max_demand` specifies the
   maximum amount of events that must be in flow while the `:min_demand`
-  specifies the minimum threshold to trigger for more demand. For
-  example, if `:max_demand` is 1000 and `:min_demand` is 750,
-  the consumer will ask for 1000 events initially and ask for more
-  only after it processes at least 250.
+  specifies the minimum threshold to trigger for more demand. The consumer
+  will never ask the producer for more than `:max_demand` events at a time.
+  When the producer emits enough events for the consumer's demand to go
+  down to `:min_demand`, only then the consumer will ask for more events
+  (in other words, it will "send more demand upstream").
+
+  An example with numbers will help clarify this. Say that `:max_demand` is
+  `1000` and `:min_demand` is `750`. Initially, the consumer asks for `1000`
+  events to the producer. Say that the producer producers `100` events at a time.
+  The first "batch" of `100` events goes to the consumer and the consumer's demand
+  goes down to `900`. Another batch is produced and the demand goes down to `800`.
+  At this point, the producer has not been asked for more events yet since the demand
+  didn't go below `:min_demand` yet. When the producer produces the next batch of
+  `100` events, the consumer will process `50` events and the demand reaches the
+  minimum of `750`. The consume sends `250` demand upstream (which is up to `:max_demand`).
+  of `250` to reach `:min_demand` again, and then consume the `50` events left.
```

Comment:
```suggestion
  of `250` to reach `:min_demand` again, and then consume the `50` events remaining.
```

Nitpick in favor of using a less ambiguous word. :)